server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -176,12 +176,13 @@ public MapperService(
             indexSettings,
             idFieldMapper
         );
+        this.mappingParserContext = this.mappingParserContextSupplier.get();
```

Comment:
Do we need to hold on to this or can we re-use the supplier on-demand?

Reply:
Indeed, we can use the supplier on-demand as it not used often (e.g. on index creation, on parsing runtime mappings during search etc), addressed in 42201db7d56f9ff2fe83c77c8b56807b3e8805cb